test/NuGet.DependencyResolver.Core.Tests/ResolverFacts.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.DependencyResolver.Core.Tests
+{
+    public class ResolverFacts
+    {
+        [Fact]
+        public void FasterProviderReturnsResultsBeforeSlowOnesIfExactMatchFound()
+        {
+            // A 
+            var slowProvider = new TestProvider(TimeSpan.FromSeconds(2));
```

Comment:
Whaaaaat? Do we really need to keep running this 2sec test every time we build just to verify that we take the fastest result? This seems like a lot of busy work during every build.